content/en/docs/concepts/cluster-administration/manage-deployment.md

```diff
@@ -14,7 +14,19 @@ You've deployed your application and exposed it via a service. Now what? Kuberne
 
 ## Organizing resource configurations
 
-Many applications require multiple resources to be created, such as a Deployment and a Service. Management of multiple resources can be simplified by grouping them together in the same file (separated by `---` in YAML). For example:
+Many applications require multiple resources to be created, such as a Deployment and a Service. Management of multiple resources can be simplified by grouping them together in the same file.
+Techniques for grouping together resources in YAML include separating:
+
+ - Different resource types with hyphens (`---`)
+ - Similar resource types as items in a list (such as `PodList`, `ServiceList`, or `SecretList`)
+ - Different resource types as items in a list (using `List`)
```

Comment:
This doesn't make sense to me here. When you submit objects to the Kubernetes API, you submit them one by one and never as a collection.

Reply:
Right. As far as I can tell, `List` (a client side invention) or `XyzList` (a server side abstraction) are both read-only. As a user, you are not supposed to create resources in using a `XyzList` resource type. Such an operation will need to be handled as a POST to the `xyzs` endpoint. However, you can only send a `GET` (translated to list operation) or a `DELETE` (translated to a deletecollection operation).

Reply:
So this does indeed work.

```
cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: ServiceList
items:
- apiVersion: v1
  kind: Service
  metadata:
    name: list-service-test
  spec:
    ports:
    - protocol: TCP
      port: 80
    selector:
      app: list-deployment-test
- apiVersion: v1
  kind: Service
  metadata:
    name: list-service-test2
  spec:
    ports:
    - protocol: TCP
      port: 8080
    selector:
      app: list-deployment-test2
EOF
service/list-service-test created
service/list-service-test2 created
```

The question is is it supposed to. My guess is that it's unintentional and not something we want users to do.

@lavalamp @deads2k - we talked about this a few months ago and we're moving towards documenting `kind: List` being used in kubectl for mixed resources. Any thoughts?

I think a section in https://kubernetes.io/docs/reference/using-api/api-concepts/ might cover what we need to accomplish here. Call out that the API returns lists of resources and kubectl uses an internal type of `kind: List`.

Reply:
Thanks @eddiezane . I'd be glad to just add a description of `kind: List` and the concrete list types to the api-concepts page and remove the other stuff I wrote.

Reply:
This is a kubectl feature, it recurses into lists (or `---` separated YAML). Both ways (all 3 if you count untyped list separately) are equivalent. It may even handle nested lists, I don't recall.

I don't think it's particularly weird for a user to use a typed version of a list if they're only making objects of that type. I wouldn't necessarily recommend it, but since it works already I definitely don't think it should be changed in a way that breaks people doing this.

Personally, I use `---` separated YAML. Users programmatically emitting JSON will find one of the List solutions more convenient, I would guess.

Reply:
@lavalamp So, would you suggest we not document `kind: ServiceList` as @eddiezane illustrated above (and `kind: List` as well)? If you think it's better to just describe these things in the context of the API, I can withdraw that content.

Reply:
I haven't read the context, sorry-- IMO this should be documented somewhere.

Reply:
@lavalamp If you just want to glance at what I did, see [Organizing resource configurations](https://deploy-preview-28608--kubernetes-io-main-staging.netlify.app/docs/concepts/cluster-administration/manage-deployment/). I split that section it into three parts to show how to separate resources in a yaml file using three hyphens, `ServiceList` (or similar), and `List`. Previously, it only showed the three hyphen approach.

Reply:
I agree with the people saying this is the wrong place for this documentation. But clearly it should be documented somewhere, because it seems to be uncommon knowledge :) It's just going to confuse people if they encounter it here, unfortunately.

Reply:
Okay. I'll drop what I wrote and just add a section on lists to the location @eddiezane mentioned ([api-concepts](https://kubernetes.io/docs/reference/using-api/api-concepts/)). Thanks!

Reply:
I removed my changes to this file.